### ISSUE TITLE
libnacl-dev package in salt's ubuntu repos conflicts with upstream package in universe

### ISSUE THREAD

**Issue by wryfi** [saltstack/salt#44872]
### Description of Issue/Question

The saltstack apt repositories for ubuntu contain a package of libnacl-dev-20110221-4.1, with an md5 of 5d9ee160847b925f459f1208e8dcd386. The upstream ubuntu repositories also have [libnacl-dev-20110221-4.1](https://packages.ubuntu.com/xenial/libnacl-dev) in universe, with an md5 of 5736a59aa1ad1b539c1dc1a4a692e722.

The result is that any system that has ubuntu universe and saltstack repos enabled in their apt sources has two packages with the same name and version, but different contents, available for install. I'm not really sure which one will get installed in this situation, or if it is even deterministic. 

I checked both trusty and xenial saltstack repos, and found this package there, but did not yet check the salt repos for other debian/ubuntu distros.

If the intention is to simply mirror the package for users who don't have universe enabled, then the saltstack repo should contain the exact same package as the upstream universe repo. If saltstack's package is substantively different from upstream, then it should have a different name or version number so that apt can distinguish between the packages. 

The current situation -- publishing a package with the same name and version, but different contents, as a package in an official upstream repo -- violates Debian packaging policy and should be avoided.

### Setup
/etc/apt/sources.list (also tested with xenial):
```
deb http://us.archive.ubuntu.com/ubuntu/ trusty universe
deb http://repo.saltstack.com/apt/ubuntu/14.04/amd64/latest trusty main
```

### Steps to Reproduce Issue

```
$ apt-get update
  ...
$ apt-cache madison libnacl-dev
libnacl-dev | 20110221-4.1 | http://us.archive.ubuntu.com/ubuntu/ trusty/universe amd64 Packages
libnacl-dev | 20110221-4.1 | http://repo.saltstack.com/apt/ubuntu/14.04/amd64/latest trusty/main amd64 Packages
```

### Versions Report
n/a

**Comment by gtmanfred** [comment]
@dmurphy18 can you take a look at this when you get a minute?

Thanks,
Daniel

**Comment by dmurphy18** [comment]
@wryfi different md5 checksum's does not make for different contents but merely reflects the different machines that the packages have been built on.  For example: apt caching can show different md5 issues for the same package in different releases, due to internal infrastructure build machines changes even though the source code used to build the packages is unchanged.

See url: 
https://groups.google.com/forum/#!msg/salt-users/epIfW2yIn5g/wmoWAKEVAwAJ

**Comment by wryfi** [comment]
Why are you rebuilding a package that already exists and putting it in a place where it conflicts with an upstream package?

If the package isn't different from upstream, you should just use the upstream package.

If the package is different from upstream, it should get a different name and/or version.

**Comment by dmurphy18** [comment]
@wryfi The reason to build a package is due to it not being present in earlier versions of the OS even if it  is now available, For example: Debian 8.0 does not contain packages which are available from Debian 8.9.  However you still need to keep those missing packages even if they are available from later releases since you cannot expect also users to upgrade their OS versions. They may well have valid unknown reasons for upgrading.  

Also packages might migrate from one OS repo to another, for example on Redhat some packages were only available from update repo in earlier versions of Redhat 7, but in later releases migrated to the base repo.  There are SaltStack customers which do not use the update repo (they have their reasons), hence certain packages which were available from update repo had to be provided, since their upgrade to later OS releases may not occur or occur at some unforeseen scheduled date.  Debian has similar customer issues.
